<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>842</width>
    <height>407</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Ra</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSplitter" name="splitterH">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="DebugTreeWidget" name="debugTree">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
      <widget class="QSplitter" name="splitterV">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="FieldWidget" name="field"/>
       <widget class="LogWidget" name="log"/>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>842</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Game">
    <property name="title">
     <string>&amp;Game</string>
    </property>
    <addaction name="actionEnableTransceiver"/>
    <addaction name="actionFlipSides"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuVideo">
    <property name="title">
     <string>Logging</string>
    </property>
    <addaction name="actionRecord"/>
   </widget>
   <widget class="QMenu" name="menuTesting">
    <property name="title">
     <string>Demo</string>
    </property>
    <addaction name="actionSimulator"/>
    <addaction name="actionInternalReferee"/>
    <addaction name="actionInputDevices"/>
    <addaction name="actionAutoPause"/>
   </widget>
   <addaction name="menu_Game"/>
   <addaction name="menuTesting"/>
   <addaction name="menuVideo"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolCommands">
   <property name="windowTitle">
    <string>Commands</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionEnableTransceiver"/>
   <addaction name="actionChargeKicker"/>
   <addaction name="actionFlipSides"/>
   <addaction name="actionSideChangeNotify"/>
  </widget>
  <widget class="QDockWidget" name="dockReferee">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Internal RefBox</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="RefereeWidget" name="referee"/>
  </widget>
  <widget class="QDockWidget" name="dockSimulator">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Simulator</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="SimulatorWidget" name="simulator"/>
  </widget>
  <widget class="QDockWidget" name="dockInput">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Input devices</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="InputWidget" name="input"/>
  </widget>
  <widget class="QDockWidget" name="dockVisualization">
   <property name="windowTitle">
    <string>Visualizations</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="VisualizationWidget" name="visualization"/>
  </widget>
  <widget class="QToolBar" name="toolTesting">
   <property name="windowTitle">
    <string>Testing</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSimulator"/>
   <addaction name="actionInternalReferee"/>
   <addaction name="actionInputDevices"/>
  </widget>
  <widget class="QToolBar" name="toolConfig">
   <property name="windowTitle">
    <string>Configuration</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionConfiguration"/>
   <addaction name="actionPlotter"/>
   <addaction name="actionRecord"/>
   <addaction name="actionSaveBacklog"/>
   <addaction name="actionRobotParameters"/>
  </widget>
  <widget class="QDockWidget" name="dockTiming">
   <property name="windowTitle">
    <string>Timing</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="TimingWidget" name="timing"/>
  </widget>
  <widget class="QDockWidget" name="dockRobots">
   <property name="windowTitle">
    <string>Robots</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="RobotSelectionWidget" name="robots"/>
  </widget>
  <widget class="QDockWidget" name="dockOptions">
   <property name="windowTitle">
    <string>Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="OptionsWidget" name="options"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionPlotter">
   <property name="text">
    <string>&amp;Plotter</string>
   </property>
  </action>
  <action name="actionEnableTransceiver">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Enable &amp;transceiver</string>
   </property>
   <property name="toolTip">
    <string>Enable transceiver</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionDisableTransceiver">
   <property name="text">
    <string>Disable transceiver</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionFlipSides">
   <property name="text">
    <string>&amp;Flip sides</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionRecord">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Record file</string>
   </property>
   <property name="toolTip">
    <string>Record File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionChargeKicker">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Charge kicker</string>
   </property>
   <property name="toolTip">
    <string>Charge kicker</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionSimulator">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simulator</string>
   </property>
   <property name="toolTip">
    <string>Simulator</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionInternalReferee">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Internal Referee</string>
   </property>
   <property name="toolTip">
    <string>Internal Referee</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionInputDevices">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Input devices</string>
   </property>
   <property name="toolTip">
    <string>Input devices</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionConfiguration">
   <property name="text">
    <string>Configuration</string>
   </property>
  </action>
  <action name="actionRobotParameters">
   <property name="text">
    <string>Robot parameters</string>
   </property>
  </action>
  <action name="actionSideChangeNotify">
   <property name="text">
    <string>Check for changed sides!</string>
   </property>
   <property name="font">
    <font>
     <pointsize>18</pointsize>
    </font>
   </property>
  </action>
  <action name="actionAutoPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause, if unfocused</string>
   </property>
  </action>
  <action name="actionSaveBacklog">
   <property name="text">
    <string>60s</string>
   </property>
   <property name="toolTip">
    <string>Save Backlog</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RefereeWidget</class>
   <extends>QWidget</extends>
   <header>refereewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SimulatorWidget</class>
   <extends>QWidget</extends>
   <header>simulatorwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>InputWidget</class>
   <extends>QWidget</extends>
   <header>inputwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>VisualizationWidget</class>
   <extends>QWidget</extends>
   <header>widgets/visualizationwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TimingWidget</class>
   <extends>QWidget</extends>
   <header>widgets/timingwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>RobotSelectionWidget</class>
   <extends>QWidget</extends>
   <header>robotselectionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FieldWidget</class>
   <extends>QGraphicsView</extends>
   <header>widgets/fieldwidget.h</header>
  </customwidget>
  <customwidget>
   <class>LogWidget</class>
   <extends>QPlainTextEdit</extends>
   <header>widgets/logwidget.h</header>
  </customwidget>
  <customwidget>
   <class>DebugTreeWidget</class>
   <extends>QTreeView</extends>
   <header>widgets/debugtreewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimulator</sender>
   <signal>toggled(bool)</signal>
   <receiver>dockSimulator</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>802</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInternalReferee</sender>
   <signal>toggled(bool)</signal>
   <receiver>dockReferee</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>39</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInputDevices</sender>
   <signal>toggled(bool)</signal>
   <receiver>dockInput</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>802</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
